inc/saiport.h

```diff
@@ -91,6 +91,60 @@ typedef struct _sai_port_oper_status_notification_t
 
 } sai_port_oper_status_notification_t;
 
+/**
+ * @brief Attribute data for #SAI_PORT_ATTR_ERROR_EVENT
+ */
+typedef enum _sai_port_error_event_t
+{
+    /** Unknown */
+    SAI_PORT_ERROR_EVENT_UNKNOWN,
+
+    /** Remote-fault */
+    SAI_PORT_ERROR_EVENT_REMOTE_FAULT,
+
+    /** Local-fault */
+    SAI_PORT_ERROR_EVENT_LOCAL_FAULT,
+
+    /** Pre-emphasis failed */
+    SAI_PORT_ERROR_EVENT_PREEMPHASIS_FAILED,
+
+    /** FEC set failed */
+    SAI_PORT_ERROR_EVENT_FEC_FAILED,
+
+    /** Speed set failed */
+    SAI_PORT_ERROR_EVENT_SPEED_FAILED,
+
+    /** Interface type set failed */
+    SAI_PORT_ERROR_EVENT_IF_TYPE_FAILED,
+
+    /** Media type set failed */
+    SAI_PORT_ERROR_EVENT_MEDIA_TYPE_FAILED,
+
+    /** Link Training set failed */
+    SAI_PORT_ERROR_EVENT_LINK_TRAINING_FAILED,
```

Comment:
@JaiOCP can event notification provide the data to know if these error are currently asserted or not